### PR TITLE
Add Search button to tabbar in home screen

### LINKED ISSUE #6846: Search icon on new tab toolbar
### Vision

As a user, I want a button on the bottom toolbar to start a search so that I can access the search bar without stretching my thumb across to the top of the screen

### Acceptance Criteria

* Search icon added to the bottom toolbar in portrait ([mock](https://app.abstract.com/share/32f6c003-3908-4ca5-81e7-37137c04eee3?collectionLayerId=f5518f50-f1e3-4e72-92ac-bd51a887eaca&mode=design))
* Tracking adding for number of clicks on the search icon and the searchbar from the homescreen to compare usage
  * Tracking must follow privacy policy
  * Should confirm with data science if this should go to Amplitude or just internal telemetry

### Not included

* The icon should only be included on the toolbar in the new tab screen. It should not appear in the toolbar when scrolling a webpage
* A/B test not needed as success will be measured based on clicks of new button
* This does not include iPad, as there is no bottom toolbar

**Not Included**

* This is not needed in landscape mode, so changes are only needed for portrait

### PR DESCRIPTION
**Pull request by joeyg**
Fixes #6846. This PR swaps out the disabled button for a search icon when the user is on the home screen. @garvankeeley not sure who to tag on the review.

Simulator Screen Shot - iPhone 8 - 2020-06-24 at 21 52 22

### CODE REVIEW

**Comment by joeyg** [comment]
This one seems a bit flaky - going to try and make the icon only show up for the Home screen instead of using `isWebPage`

**Comment by garvankeeley** [comment]
Thanks for looking at this! I am swamped and have little/no familiarity with this issue, I'll see if @dnarcese or @nbhasin2 have cycles sooner than me.

**Comment by nbhasin2** [comment]
Picking this one on to review.

**Comment by joeyg** [comment]
@dnarcese or @nbhasin2 I updated this PR to include some tests. Let me know what you think.

**Comment by nbhasin2** [comment]
Sounds good @joeyg, going to review this tomorrow.

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift [hunk 1]
```diff
@@ -184,5 +184,13 @@ extension BrowserViewController: TabToolbarDelegate, PhotonActionSheetProtocol {
             self.present(backForwardViewController, animated: true, completion: nil)
         }
     }
+
+    func tabToolbarDidPressSearch(_ tabToolbar: TabToolbarProtocol, button: UIButton) {
+        focusLocationTextField(forTab: tabManager.selectedTab)
+    }
+
+    func tabToolbarDidLongPressSearch(_ tabToolbar: TabToolbarProtocol, button: UIButton) {
```
Comment (nbhasin2) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift:192]: Do we need this while we already have `func tabToolbarDidLongPressReload(_ tabToolbar: TabToolbarProtocol, button: UIButton) {` which gets called when you long press the search icon?

File: Client/Frontend/Browser/TabToolbar.swift [hunk 2]
```diff
@@ -46,15 +55,40 @@ open class TabToolbarHelper: NSObject {
 
     let ImageReload = UIImage.templateImageNamed("nav-refresh")
     let ImageStop = UIImage.templateImageNamed("nav-stop")
+    let ImageSearch = UIImage.templateImageNamed("search")
 
-    var loading: Bool = false {
-        didSet {
-            if loading {
+    func setMiddleButtonState(_ state: MiddleButtonState) {
```
Comment (nbhasin2) [inline Client/Frontend/Browser/TabToolbar.swift:60]: Could set access control of `setMiddleButtonState` method to `fileprivate setMiddleButtonState`

File: Client/Frontend/Browser/TabToolbar.swift [hunk 3]
```diff
@@ -46,15 +55,40 @@ open class TabToolbarHelper: NSObject {
...
+            case .stop:
                 toolbar.stopReloadButton.setImage(ImageStop, for: .normal)
                 toolbar.stopReloadButton.accessibilityLabel = NSLocalizedString("Stop", comment: "Accessibility Label for the tab toolbar Stop button")
+            case .search:
+                toolbar.stopReloadButton.setImage(ImageSearch, for: .normal)
+                toolbar.stopReloadButton.accessibilityLabel = NSLocalizedString("Search", comment: "Accessibility Label for the tab toolbar Search button")
+        }
+    }
+
+    var loading: Bool = false {
```
Comment (nbhasin2) [inline Client/Frontend/Browser/TabToolbar.swift:74]: Could we also update access control of `loading` to `fileprivate var loading`

File: Client/Frontend/Browser/TabToolbar.swift [hunk 4]
```diff
@@ -46,15 +55,40 @@ open class TabToolbarHelper: NSObject {
...
+                if loading {
+                    setMiddleButtonState(.stop)
+                } else {
+                    setMiddleButtonState(.reload)
+                }
+            }
+        }
+    }
+
+    var isSearch: Bool = false {
```
Comment (nbhasin2) [inline Client/Frontend/Browser/TabToolbar.swift:86]: Update access control of `isSearch` to `fileprivate var isSearch`

**Comment by nbhasin2** [comment]
@joeyg Thanks for taking time to look at the search icon improvement. Requested some changes but overall it looks good :) 
Going to take a deeper look at the tests to see if there is a way to avoid adding empty stubs from protocol. 

Did you get a chance to test it on a smaller screen like iPhone 5s?

**Comment by joeyg** [comment]
iPhone 5s
Simulator Screen Shot - iPhone 5s - 2020-06-27 at 13 34 54

**Comment by nbhasin2** [comment]
@joeyg Turns out making it a fileprivate variable will have issues with test 😞 
Guess, we'll have to change its access control back so that we can run the test. 

image

**Pull request merged**